liana/method/sc/_liana_pipe.py

```diff
@@ -528,3 +535,151 @@ def _trimean(a, axis=0):
                     q=[0.25, 0.5, 0.5, 0.75],
                     axis=axis),
         axis=axis)
+        
+def _cluster_mean(lr_res, adata):
+    """
+    Evaluate the mean of the expression matrix for each cluster of cells.
+    Considers all the genes maintained in the expression matrix at this poit.
+
+    Parameters
+    ----------
+    lr_res 
+        pd.DataFrame (Dataframe of results for each ligand-receptor pair)
+    adata 
+        anndata.AnnData (Annotated data object. For scSeqComm must not be filtered by gene expression.)
+
+    Returns 
+    ----------
+    lr_res (returns the ligand-receptor pair dataframe with a new column for the mean of the source and target cluster.)
+    
+    """
+    labels = adata.obs['@label'].cat.categories
+    dedict = {label: None for label in labels}
+    for label in labels:
+        temp = adata[adata.obs['@label'].isin([label])]
+        dedict[label] = np.mean(temp.X.A,axis=None)
+
+    lr_res['source_cluster_mean'] = lr_res['source'].map(dedict)
+    lr_res['target_cluster_mean'] = lr_res['target'].map(dedict)
+    return lr_res
+
+def _cluster_sd(lr_res, adata):
+    """
+    Evaluate the standard deviation of the expression matrix for each cluster of cells.
+    Considers all the genes maintained in the expression matrix at this poit.
+
+    Parameters
+    ----------
+    lr_res
+        pd.DataFrame (Dataframe of results for each ligand-receptor pair)
+    adata
+        anndata.AnnData (Annotated data object. For scSeqComm must not be filtered by gene expression.)
+
+    Returns 
+    ----------
+    lr_res (returns the ligand-receptor pair dataframe with a new column for the standard deviation of the source and target cluster.)
+    
+    """
+    labels = adata.obs['@label'].cat.categories
+    dedict = {label: None for label in labels}
+    for label in labels:
+        temp = adata[adata.obs['@label'].isin([label])]
+
+        dedict[label] = np.std(temp.X.A,axis=None)
+    lr_res['source_cluster_std'] = lr_res['source'].map(dedict)
+    lr_res['target_cluster_std'] = lr_res['target'].map(dedict)
+    
+    return lr_res
+
+def _cluster_counts(lr_res, adata):
+    """
+    Evaluate the cardinality of the expression matrix for each cluster of cells.
+
+    Parameters
+    ----------
+    lr_res
+        pd.DataFrame (Dataframe of results for each ligand-receptor pair)
+    adata
+        anndata.AnnData (Annotated data object. For scSeqComm must not be filtered by gene expression.)
```

Comment:
This is incompatible with how LIANA is set up. Essentially, it will remove genes and cells with no counts.

Also, most people would probably prefer to use the filtered object by the time they reach CCC :)